oak/server/rust/oak_runtime/src/node/grpc/server/mod.rs

```diff
@@ -258,6 +267,54 @@ impl Service<http::Request<hyper::Body>> for HttpRequestHandler {
     }
 }
 
+impl From<OakLabelError> for tonic::Status {
+    fn from(v: OakLabelError) -> Self {
+        match v {
+            OakLabelError::MissingLabel => tonic::Status::invalid_argument("Missing Oak Label"),
+            OakLabelError::MultipleLabels => tonic::Status::invalid_argument("Multiple Oak Labels"),
+            OakLabelError::InvalidLabel => tonic::Status::invalid_argument("Invalid Oak Label"),
+        }
+    }
+}
+
+enum OakLabelError {
+    MissingLabel,
+    MultipleLabels,
+    InvalidLabel,
+}
+
+/// Returns the [`Label`] defined as part of the the metadata of an incoming gRPC request.
+///
+/// Returns an error if there is not exactly one label specified by the caller.
+fn get_oak_label(metadata_map: &MetadataMap) -> Result<Label, OakLabelError> {
```

Comment:
Could this function be used for gRPC client too?
If so, should it be in the `grpc/mod.rs`
WDYH?

Reply:
I don't think it can be used there, there are no incoming requests in the gRPC client, right?